indicator-nara/031-berekening.Rmd

```diff
@@ -0,0 +1,707 @@
+#gemiddeld is het 30 dagen
+
+#berekenen we een drempelwaarde over het totaal aantal metingen over 30 jaar. Is dit dan ook nog zo ?
+absperc.30 <- ruwedata %>% 
+  mutate(jaar = year(dag)) %>% 
+  filter(dplyr::between(jaar,1985,2014)) %>% 
+  group_by(meetpunt) %>% 
+  summarise(p01 = quantile(meting_TAW, 1/100),
+            p05 = quantile(meting_TAW, 5/100),
+            p10 = quantile(meting_TAW, 10/100),
+            p30 = quantile(meting_TAW, 30/100),
+            p50 = quantile(meting_TAW, 50/100),
+            p70 = quantile(meting_TAW, 70/100),
+            p90 = quantile(meting_TAW, 90/100),
+            p95 = quantile(meting_TAW, 95/100),
+            p99 = quantile(meting_TAW, 99/100)
+            ) %>% 
+  ungroup()
+
+d.check.30 <- ruwedata %>% 
+  filter(dplyr::between(year(dag),1985,2014)) %>% 
+  inner_join(absperc.30) %>% 
+  mutate(jaar = year(dag)) %>% 
+  group_by(jaar, meetpunt) %>% 
+  summarise(aantaldroog = sum(ifelse(meting_TAW < p05,1,0 ))) %>% 
+  ungroup() %>% 
+  group_by(meetpunt) %>% 
+  summarise(gemdroog = mean(aantaldroog)/20)
+#dit klopt dus: netjes 18.25 (behalve voor een meetpunt met slechts 29 meetjaren). Dit moest mathematisch wel kloppen.
+
+#Dit betekent dat het verschil wel degelijk komt door eerst jaargemiddelden te berekenen en hiervan het gemiddelde te nemen.
+#het feit dat het 30-jaargemiddelde duidelijk hoger is dan 18.25 (nl. ong. 30) ligt aan het feit dat een nat jaar nooit meer dan 18 dagen kan afwijken van een normaal jaar, terwijl een droog jaar tot wel 365 dagen kan afwijken => (28jaar * 18.25 + 1*365 + 1*0)/30 > 18.25
+
+```
+
+```{r schrikkeljaren, include=FALSE}
+
+schrikkeljaar <- ruwedata %>% 
+  mutate(jaar = year(dag)) %>%   
+  filter(meetpunt == 
+           ruwedata %>% 
+           dplyr::select(meetpunt) %>% 
+           head(1) %>% 
+           dplyr::pull(meetpunt)
+         ) %>% 
+  filter(day(dag) == 29, month(dag) == 2) %>% 
+  distinct(jaar) %>% 
+  arrange(jaar) %>% 
+  dplyr::pull(jaar)
+
+```
+
+```{r gwgroups, include=FALSE, eval = !file.exists(file.path("cache", "percentielen.rds" ))}
+#herijking grondwatertype-groepen, want er blijkt nog nauwelijks een verband te bestaan tussen de amplitude van een pb en de gw-typegroep waartoe het wordt gerekend
+absperc_gw <- absperc %>% 
+  inner_join(tubes_indicator %>% 
+               mutate(groep3 = ifelse(groupnr == 1, 2, groupnr)) %>% 
+               dplyr::select(-groupnr), by = c("meetpunt" = "loc_code")) %>% 
+  mutate(ampl = p99 - p01,
+         groep3n = case_when(
+           ampl < 0.25 ~2,
+           ampl < 0.4 ~3,
+           TRUE ~4),
+         meetpunt = factor(meetpunt)
+         )
+
+
+#wegschrijven in een RDS-bestand
+saveRDS(absperc_gw,file.path("cache", "percentielen.rds" ))
+
+write_vc(absperc_gw, file.path("data", "result", "percentielen-1985-2014"), sorting = c("meetpunt"), strict = FALSE)
```

Comment:
Use `strict = FALSE` only when the definition of the variable change. You might want to set the root as well.

```suggestion
write_vc(absperc_gw, file.path("data", "result", "percentielen-1985-2014"), sorting = c("meetpunt"))
```

Reply:
OK, i skipped the strict-argument and added a root-argument (more secure use)